File: state/c-sharp/InitializeState.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace State
{
    // Farklı durumların meydana geldiği sınıf.
    // UML diyagramındaki State (VendingMachineState) yapısından türer.
    // UML diyagramındaki ConcreteState yapısına denk gelmektedir.
    class InitializeState : VendingMachineState
    {
        // Başlangıç durumu ayarları yapıldıktan sonra WaitingState (hazır) durumuna geçer.
        public override void HandleState(VendingMachine context)
        {
            Console.WriteLine("InitializeState: Ön hazırlıklar yapılıyor.");

            context.SetState(new WaitingState());
        }
    }
}

File: state/typescript/initialize-state.ts
import {VendingMachineState} from "./vending-machine-state";
import {VendingMachine} from "./vending-machine";
import {WaitingState} from "./waiting-state";

// Farklı durumların meydana geldiği sınıf.
// UML diyagramındaki State (VendingMachineState) yapısından türer.
// UML diyagramındaki ConcreteState yapısına denk gelmektedir.
export class InitializeState extends VendingMachineState {
  // Başlangıç durumu ayarları yapıldıktan sonra WaitingState (hazır) durumuna geçer.
  public handleState(context: VendingMachine): void {
    console.log('InitializeState: Ön hazırlıklar yapılıyor.');

    context.setState(new WaitingState());
  }
}
